src/collective/siteimprove/browser/viewlets.py

```diff
@@ -13,9 +13,21 @@ def update(self):
         super(SiteimproveJavascriptViewlet, self).update()
 
         self.token = None
+        self.recheck = False
 
         # lookup token
         registry = getUtility(IRegistry)
         siteimprove_registry = registry.forInterface(ISiteimproveSchema, False)
         if siteimprove_registry and siteimprove_registry.token:
-            self.token = siteimprove_registry.token
\ No newline at end of file
+            self.token = siteimprove_registry.token
+
+        # check if a publish event proceeded this, if yes, inject site improve recheck js code
+        cookie = self.request.cookies.get("SI-Published", None)
+        if cookie:
+            self.recheck = True
+            # clear the cookie
+            # trying all sorts of way to cleart the cookie. Not working!
+            # once set, the cooke does not go away! :(
+            del self.request.cookies['SI-Published']
+            self.request.response.expireCookie("SI-Published")
+            self.request.cookies["SI-Published"] = False
```

Comment:
Does this do anything? I think the expireCookie call should suffice.

Reply:
every time I reload my test page, cookie is True and so this if statement is executed.